website/django/data/views.py

```diff
@@ -148,3 +157,299 @@ def sanitise_term(term):
+            var_resp.start = brca_variant[j]
+            var_resp.end = brca_variant["Hg36_End"]
+            continue
+        if j == "Hg36_End" and (genRefer == "hg36"): continue
+
+        if j == "Hg37_Start" and (genRefer == "hg37"):
+            var_resp.start = brca_variant[j]
+            var_resp.end = brca_variant["Hg37_End"]
+            continue
+        if j == "Hg37_End" and (genRefer == "hg37"): continue
+
+        if j == "Hg38_Start" and (genRefer == "hg38"):
+            var_resp.start = brca_variant[j]
+            var_resp.end = brca_variant["Hg38_End"]
+            continue
+        if j == "Hg38_End" and (genRefer == "hg38"): continue
+
+        if j == "Synonyms":
+            Names = [i for i in str(brca_variant[j]).split(",")]
+            for i in Names:
+                var_resp.names.append(i)
+        else:
+            var_resp.info[str(j)].append(brca_variant[j])
+    return var_resp
+################## Translator function END ##############################
+
+#### Auxiliary function which validates requests ######
+def validate_request(request):
+    if not request.body:
+        return HttpResponseBadRequest(json.dumps(ErrorMessages['emptyBody']), content_type="application/json")
+    else:
+        request_dict = json.loads(request.body)
+        if not request_dict.get("variantSetId"):
+            return HttpResponseBadRequest(json.dumps(ErrorMessages['VariantSetId']), content_type="application/json")
+        elif not request_dict.get('referenceName'):
+            return HttpResponseBadRequest(json.dumps(ErrorMessages['referenceName']), content_type="application/json")
+        elif not request_dict.get('start')  :
+            return HttpResponseBadRequest(json.dumps(ErrorMessages['start']), content_type="application/json")
+        elif not request_dict.get('end') :
+            return HttpResponseBadRequest(json.dumps(ErrorMessages['end']), content_type="application/json")
+        else:
+            return None
+
+def validate_varsetreq(request):
+    if not request.body:
+        return HttpResponseBadRequest(json.dumps(ErrorMessages['emptyBody']), content_type="application/json")
+    else:
+        request_dict = json.loads(request.body)
+        if not request_dict.get("datasetId"):
+            return HttpResponseBadRequest(json.dumps(ErrorMessages['datasetId']), content_type="application/json")
+        else:
+            return None
+########## Auxiliary function END ############
+
+######### .../variants/<variant id> method ########
+@require_http_methods(["GET"])
+def get_var_by_id(request, variant_id):
+    if not variant_id:
+        return HttpResponseBadRequest(json.dumps(ErrorMessages['variantId']), content_type="application/json")
+    else:
+        gen_coor_and_id = variant_id
+        gen_coor, v_id = gen_coor_and_id.split("-")
+        if gen_coor in SetIds:
+            DbResp = Variant.objects.values()
+            resp1 = DbResp.get(id=int(v_id))
```

Comment:
Need to test if v_id exists too. What happens when we look for variant hg38-123123abc